e2e/internal/kubeclient/deploy.go

```diff
@@ -255,8 +255,8 @@ retryLoop:
 	}
 }
 
-// WaitForLoadBalancer waits until the given service is configured with an external IP and returns it.
-func (c *Kubeclient) WaitForLoadBalancer(ctx context.Context, namespace, name string) (string, error) {
+// WaitForService waits until the given service is configured with an external IP and returns it.
+func (c *Kubeclient) WaitForService(ctx context.Context, namespace, name string, loadBalancer bool) (string, error) {
```

Comment:
> Is there actually a window where the ClusterIP is unassigned?

I don't think there is.

> Otherwise, we could just directly use the client and avoid the `loadBalancer == false` case.

`WaitForService` (or formerly `WaitForLoadBalancer`) doesn't just get the IP, it also waits for the port to be reachable, so we can't just replace the whole function by using the client directly to query the IP.

Reply:
That's what I tried to document in https://github.com/edgelesssys/contrast/pull/865#issuecomment-2340140883